scripts/linkchecker.py

```diff
@@ -426,31 +435,37 @@ def parse_arguments():
 
     Result is returned and saved into global variable ARGS.
     """
-    parser = argparse.ArgumentParser(description="Links checker for docs.")
-    parser.add_argument("-l", dest="lang", default="en", metavar="<LANG>",
-                        help=("two letter language code, e.g. 'zh'. "
-                              "(default='en')"))
-    parser.add_argument("-v", dest="verbose", action="store_true",
+    global PARSER
+
+    PARSER = argparse.ArgumentParser(description="Links checker for docs.")
+    PARSER.add_argument("-v", dest="verbose", action="store_true",
                         help="switch on verbose level")
-    parser.add_argument("-f", dest="filter", default="/docs/**/*.md",
-                        metavar="<FILTER>",
-                        help=("File pattern to scan, e.g. '/docs/foo.md'. "
-                              "(default='/docs/**/*.md')"))
-    parser.add_argument("-n", "--no-color", action="store_true",
+    PARSER.add_argument("-n", "--no-color", action="store_true",
                         help="Suppress colored printing.")
+    PARSER.add_argument(dest="filter", default="content/en/docs/**/*.md",
```

Comment:
It seems that the default value for the argument `filter` (`content/en/docs/**/*.md`) is not being applied properly.

[How to reproduce]
: run `./linkchecker.py` with no arguments

[What I expected]
: `./linkchecker.py` is executed with the default value for the argument `filter` (`content/en/docs/**/*.md`).

[What actually happens]
: `./linkchecker.py` returns error that `filter` should be specified.
```
❯ ./linkchecker.py      
usage: linkchecker.py [-h] [-v] [-n] <FILTER>
linkchecker.py: error: the following arguments are required: <FILTER>
```

---

Furthermore, when I specify the `filter` as the default value (`content/en/docs/**/*.md`),
this script completes with displaying no link errors,

[on the branch [`tengqm:improve-linkchecker`](https://github.com/tengqm/website/tree/improve-linkchecker), using the `linkchecker.py` that this PR suggests]

```
❯ ./linkchecker.py content/en/docs/**/*.md

Completed link validation.
```

but actually there do exist some link errors in `content/en/docs/**/*.md`, which I can confirm with the current version of `linkchecker.py`.

[on the latest `main`, using the current version of `linkchecker.py`]
```
❯ ./linkchecker.py

...
File: ../content/en/docs/tasks/access-application-cluster/configure-access-multiple-clusters.md
    /releases/version-skew-policy/#kubectl: Link may be wrong for the anchor [within one minor version]
File: ../content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md
    /examples/service/access/Dockerfile: Examples link, skipped
File: ../content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md
    https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema: Should use relative paths
File: ../content/en/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information.md
    /concepts/storage/volumes/#downwardapi: Link may be wrong for the anchor [`downwardAPI` volume]
File: ../content/en/docs/tasks/job/coarse-parallel-processing-work-queue.md
    /examples/application/job/rabbitmq/Dockerfile: Examples link, skipped
File: ../content/en/docs/tasks/job/fine-parallel-processing-work-queue.md
    /examples/application/job/redis/Dockerfile: Examples link, skipped
    /examples/application/job/redis/rediswq.py: Examples link, skipped
    /examples/application/job/redis/Dockerfile: Examples link, skipped
File: ../content/en/docs/tasks/job/automated-tasks-with-cron-jobs.md
    /docs/concepts/workloads/controllers/garbage-collection/: Link using redirect records, should use /docs/concepts/architecture/garbage-collection/ instead
File: ../content/en/docs/doc-contributor-tools/linkchecker/README.md
    : Link may be wrong for the anchor [title]
Completed link validation.
```